File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> v;
        for(auto s:strs)
        {
            string key=s;
            sort(key.begin(),key.end());
            v[key].push_back(s);
        }
        vector<vector<string>> ans;
        // for(int i=0;i<v.size();i++)
        // {
        //     ans.push_back(v[i].second);
        // }
        //unordered_map does not support indexing
        for(auto& pair : v) {
            ans.push_back(pair.second);
        }

        return ans;
    }


};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>> mpp=new HashMap<>();
        for(String s:strs)
        {
            char[] key=s.toCharArray();
            Arrays.sort(key);
            String n=new String(key);
            if(!mpp.containsKey(n))
            {
                mpp.put(n,new ArrayList<>());
            }

            mpp.get(n).add(s);
        }

        return new ArrayList<>(mpp.values());
    }
}